### PR TITLE
Automated tests for Availability Times

### PR DESCRIPTION
**Pull request by bjss-colin**

### CODE REVIEW

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/AddNewSolution/ServiceLevelAgreements/AddServiceAvailabilityTimes.cs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+            CommonActions.PageTitle()
+                .Should()
+                .BeEquivalentTo($"Service availability times - {solution.Name}".FormatForComparison());
+
+            CommonActions.ElementIsDisplayed(CommonSelectors.Header1).Should().BeTrue();
+            CommonActions.ElementIsDisplayed(ServiceAvailabilityTimesObjects.SupportTypeInput).Should().BeTrue();
+            CommonActions.ElementIsDisplayed(ServiceAvailabilityTimesObjects.FromInput).Should().BeTrue();
+            CommonActions.ElementIsDisplayed(ServiceAvailabilityTimesObjects.UntilInput).Should().BeTrue();
+            CommonActions.ElementIsDisplayed(ServiceAvailabilityTimesObjects.ApplicableDaysInput).Should().BeTrue();
```
Comment (leerogers-bjss) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/AddNewSolution/ServiceLevelAgreements/AddServiceAvailabilityTimes.cs:50]: need to add a check in here to make sure that the delete link doesn't exist in this scenario
would suggest
```c#
CommonActions.ElementExists(ServiceAvailabilityTimesObject.DeleteLink).Should().BeFalse();
```

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/AddNewSolution/ServiceLevelAgreements/DeleteServiceAvailabilityTimes.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+using NHSD.GPIT.BuyingCatalogue.E2ETests.Objects.Common;
+using NHSD.GPIT.BuyingCatalogue.E2ETests.Utils;
+using NHSD.GPIT.BuyingCatalogue.E2ETests.Utils.TestBases;
+using NHSD.GPIT.BuyingCatalogue.EntityFramework.Ordering.Models;
+using NHSD.GPIT.BuyingCatalogue.WebApp.Areas.Admin.Controllers;
+using Xunit;
+
+namespace NHSD.GPIT.BuyingCatalogue.E2ETests.Areas.Admin.AddNewSolution.ServiceLevelAgreements
+{
+    public sealed class DeleteServiceAvailabilityTimes : AuthorityTestBase, IClassFixture<LocalWebApplicationFactory>
```
Comment (leerogers-bjss) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/AddNewSolution/ServiceLevelAgreements/DeleteServiceAvailabilityTimes.cs:15]: No test added for the cancel link being clicked

**Review by leerogers-bjss — CHANGES_REQUESTED** [review]
just 1 minor change and a missing test scenario.

looks good beyond that :)

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/AddNewSolution/ServiceLevelAgreements/AddServiceAvailabilityTimes.cs (hunk 1 above)
Comment (bjss-colin) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/AddNewSolution/ServiceLevelAgreements/AddServiceAvailabilityTimes.cs:50]: Good shout, didn't think to add it here originally but it's certainly relevant.

**Review by leerogers-bjss — APPROVED** [review]

**Pull request merged**